features/step_definitions/user_steps.rb

```diff
@@ -0,0 +1,21 @@
+Given("User visits the site") do
+  visit '/'
+
+When("User clicks on the {string} button") do |button|
+  click_on(button)
+
+Then("He should be redirected to {string}") do |string|
+  visit string
+end
+
+Then("He fills in {string} with {string}") do |field, input|
+  fill_in(field, :with => input) 
+end
+
+Then("He clicks {string}") do |string|
+  click_on(string)
+end
+
+Then("He sees message {string}") do |string|
+  page.should have_content('signup successful')
```

Comment:
Again the variable name string  is not descriptive of its purpose.
You are checking for hardcoded content when you should be checking if the value in the string variable is being displayed.